### ISSUE TITLE
Kubeadm use deprecated controller-manager and scheduller flags

### ISSUE THREAD

**Issue by yuripastushenko** [kubernetes/kubeadm#1234]
## Is this a BUG REPORT or FEATURE REQUEST?
BUG REPORT

## Versions
**kubeadm version** v1.12.2

## What happened?
When I run `kubeadm init` without any parameters -  one of steps is generating manifests for kube-scheduller (/etc/kubernetes/manifests/kube-scheduler.yaml) and kube-controller-manager (/etc/kubernetes/manifests/kube-controller-manager.yaml).
In both manifests, one of the flags that is given at startup to application is:
`--address=127.0.0.1`.

This flag is deprecated for kube scheduler - see https://kubernetes.io/docs/reference/command-line-tools-reference/kube-scheduler/#options .
And this flag is absent in options of kube-controller-manager - see https://kubernetes.io/docs/reference/command-line-tools-reference/kube-controller-manager/#options.

But those flags at this moment are still working. 

## What you expected to happen?
I expected to see flag:
`--bind-address=127.0.0.1` in both manifests (kube-controller-manager and kube-scheduler) instead of `--address=127.0.0.1`

## How to reproduce it (as minimally and precisely as possible)?
run:
`kubeadm init --dry-run`
and see those flags in output, in sections:
```
[dryrun] Would write file "/etc/kubernetes/manifests/kube-controller-manager.yaml" with content:
[dryrun] Would write file "/etc/kubernetes/manifests/kube-scheduler.yaml" with content:
 ```

**Comment by neolit123** [comment]
marking this is a good first issue.

a change has to be made here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/controlplane/manifests.go#L254
and here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/controlplane/manifests.go#L298

and in the file with the unit tests too:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/phases/controlplane/manifests_test.go

**Comment by neolit123** [comment]
@kubernetes/sig-cluster-lifecycle

**Comment by neolit123** [comment]
/assign @yagonobre 
/lifecycle active

i wanted to see comments about these address fields as we have them in the kubeadm design document as 127.0.0.1.
`0` or  `::` means default network interface, so i don't think we should remove them.
cc @luxas @kad

**Comment by Arslanbekov** [comment]
Hello!
We have the same issue like [#1235](https://github.com/kubernetes/kubeadm/issues/1235).
Any suggestions guys?

**Comment by timothysc** [comment]
We are out of time, punt to 1.14

**Comment by neolit123** [comment]
xref:
https://github.com/kubernetes/kubeadm/issues/1285
cc @luxas

**Comment by luxas** [comment]
@yagonobre any progress?

**Comment by yagonobre** [comment]
@luxas I'll send a pr tomorrow.

**Comment by yuripastushenko** [comment]
Like i told in https://github.com/kubernetes/kubeadm/issues/1235 - i think we should by default exclude this flag at all, because the default value of --bind-address is driven by kubeadm and is 0.0.0.0.
The users should have the ability to leave the default behaviour of kubeadm for this flag.

**Comment by rosti** [comment]
/lifecycle active
Due to pending PR by @MalloZup

**Comment by MalloZup** [comment]
# todos:

- [ ] adress this  https://github.com/kubernetes/kubernetes/pull/71973#issuecomment-449031692

**Comment by dannyk81** [comment]
Hello,

Was just got hit by this issue, we are deploying K8s 1.13.2 with Kubespray and set `bind-address` for both `scheduler` and `controller-manager` to `0.0.0.0`, however the deprecated `--address` is still being added to the manifest and seems to take precedence.

I noticed that this was released as part of `1.14` alpha release, but looks like it didn't make it to 1.12 or 1.13 yet.

Is it possible to add this fix on these releases too?

/cc @neolit123
